src/checks/state-checks.ts

```diff
@@ -51,18 +52,36 @@ export const stateChecks = (
     return errorMessages;
 }
 
-const getPropertyCount = ({props, state}: {
+const getPropertyCount = ({props, object}: {
     props: string[],
-    state: State,
-}) : number => {
-    return props.map((prop) => prop in state ? 1 : 0)
```

Comment:
changed this to work with a generic json object as opposed to assuming it's the State itself

the logic for fetching the props dynamically doesn't change